### DISCUSSION TITLE
Support `BigFloat`/`BigInt`?

### DISCUSSION THREAD

**Discussion by jariji** [Seelengrab/Supposition.jl#20] [Design & Ideas]
After getting annoyed by all the fiddly details when testing with normal floats, I wanted to to try testing a claim using `BigFloat`. But `Data.Floats{BigFloat}()` doesn't work.

```jl
julia> Data.Floats{BigFloat}()
ERROR: TypeError: in Floats, in T, expected T<:Union{Float16, Float32, Float64}, got Type{BigFloat}
```

**Comment by Seelengrab** [comment]
Well, those fiddly details are just what make floating point floating point :)

`BigFloat` (like `BigInt`) is currently unsupported since managing its state is quite a bit more challenging. There are not a lot of ways to construct a `BigFloat`/`BigInt`, and generating one from scratch is itself challenging. Hypothesis has this limitation too, it only supports up to 128-bit wide integers [as far as I can tell](https://github.com/HypothesisWorks/hypothesis/blob/16f5205a2050231851f55505b40d480beef62bfe/hypothesis-python/src/hypothesis/internal/conjecture/data.py#L1561-L1565) and [16/32/64-bit floating point numbers](https://github.com/HypothesisWorks/hypothesis/blob/16f5205a2050231851f55505b40d480beef62bfe/hypothesis-python/src/hypothesis/internal/conjecture/data.py#L1584-L1593).

I'm not opposed in principle to add this at some point, but it's going to require at the very least #25 . There isn't really a good way to construct a `BigInt`/`BigFloat` from the raw choicesequence, without copying those (potentially huge) numbers around a lot.
